docs/deploy.md

```diff
@@ -13,45 +13,49 @@ Let's look at how serverless applications are deployed on AWS with SAM.
 
 ### Stacks
 
-Everything is deployed through a **[CloudFormation](https://aws.amazon.com/cloudformation/) stack**. A stack is nothing more than a bunch of things that compose an application:
+Everything is deployed through a **[CloudFormation](https://aws.amazon.com/cloudformation/) stack**. A stack defines AWS resources to be created on your account. An AWS resource on CloudFormation can be anything, such as:
 
-- lambda functions
+- Lambda Functions
 - S3 buckets
-- databases
+- RDS
+- DynamoDB
+- VPC
+- Load Balancer
+- API Gateway
 
-Stacks make it easy to group those resources together: the whole stack is updated at once on deployments, and if you delete the stack all the resources inside are deleted together too. Clean and simple.
+Stacks provides an unified way to manage AWS resources. When a stack is first created, all of it's resources definitions are created. You may add additional resources later, remove existing resources or update them. You may also remove all resources at once by removing the entire stack.
 
-All of this is great except CloudFormation configuration is complex. This is where SAM helps.
+CloudFormation primary goal is to provide one unified structure to provision any AWS resource on your account. SAM on the other hand is a wraper around CloudFormation dedicated to facilitating a Lambda function deployment.
 
 ### SAM
 
 SAM offers a simpler configuration format. This is what you are using if you use Bref (the `template.yaml` file).
 
-The deployment process with SAM works like this:
+The deployment process with SAM is divided into three steps:
```

Comment:
👍 This change is great because it makes it more explicit that these steps follow each other.

We could still make it simpler I think, for example:

> Deploying with SAM is done in 3 steps: